cdk/domino_cdk/provisioners/eks/eks_iam_roles_for_k8s.py

```diff
@@ -0,0 +1,176 @@
+from typing import Dict, List
+
+import aws_cdk.aws_eks as eks
+import aws_cdk.aws_iam as iam
+from aws_cdk import core as cdk
+from aws_cdk.aws_s3 import Bucket
+
+# Permission groups
+
+s3_write_permissions = [
+    "s3:PutObject",
+    "s3:DeleteObject",
+    "s3:ListMultipartUploadParts",
+    "s3:AbortMultipartUpload",
+]
+
+s3_read_permissions = [
+    "s3:GetObject",
+]
+
+ecr_policies = {
+    "ecr_read_write_permissions": [
+        "ecr:PutImageTagMutability",
+        "ecr:StartImageScan",
+        "ecr:ListTagsForResource",
+        "ecr:UploadLayerPart",
+        "ecr:BatchDeleteImage",
+        "ecr:ListImages",
+        "ecr:DeleteRepository",
+        "ecr:CompleteLayerUpload",
+        "ecr:TagResource",
+        "ecr:DescribeRepositories",
+        "ecr:DeleteRepositoryPolicy",
+        "ecr:BatchCheckLayerAvailability",
+        "ecr:ReplicateImage",
+        "ecr:GetLifecyclePolicy",
+        "ecr:PutLifecyclePolicy",
+        "ecr:DescribeImageScanFindings",
+        "ecr:GetLifecyclePolicyPreview",
+        "ecr:CreateRepository",
+        "ecr:PutImageScanningConfiguration",
+        "ecr:GetDownloadUrlForLayer",
+        "ecr:DeleteLifecyclePolicy",
+        "ecr:PutImage",
+        "ecr:UntagResource",
+        "ecr:BatchGetImage",
+        "ecr:DescribeImages",
+        "ecr:StartLifecyclePolicyPreview",
+        "ecr:InitiateLayerUpload",
+        "ecr:GetRepositoryPolicy",
+    ]
+}
+
+s3_policies = {
+    "write_blobs_read_logs": {
+        "blobs": s3_read_permissions + s3_write_permissions,
+        "logs": s3_read_permissions,
+    },
+    "write_blobs": {
+        "blobs": s3_read_permissions + s3_write_permissions,
+    },
+    "read_logs": {
+        "logs": s3_read_permissions,
+    },
+    "read_blobs": {
+        "blobs": s3_read_permissions,
+    },
+}
+
+
+# Roles
+roles = {
```

Comment:
I'm worried we're missing: fluentd (for S3 writing), cluster-autoscaler (for node management), docker-registry (for S3 storage), nucleus (for volume snapshotting), backups (git/mongo/postgres to S3), fc-agent (optional, but a few observability things). Since this is incremental work, I think it's ok, we just need to make sure we continue to incrementally move things to pod roles.

Reply:
For now I have just added roles I'm aware of. Definitely we will have much bigger list.